components/ChainSelector.tsx

```diff
@@ -46,18 +47,19 @@ const ChainSelector = () => {
       onForkChange(option.value)
 
       router.query.fork = option.value
-      router.push(router)
+      router.replace({ query: router.query })
     },
     // eslint-disable-next-line react-hooks/exhaustive-deps
     [onForkChange],
   )
 
-  useEffect(() => {
-    if (defaultForkOption) {
-      handleForkChange(defaultForkOption)
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [defaultForkOption])
+  // TODO: don't redirect (or at least don't override address query param)
```

Comment:
Is this commented out block needed?

Reply:
my bad it shouldn't be part of the commit. it removed my query param so instead of fixing I commented it out. I will fix it before the merge.